src/authenticationService/index.js

```diff
@@ -0,0 +1,14 @@
+import * as firebase from 'firebase'
+import AuthenticationError from '~/AppErrors/AuthenticationError'
+
+const getCurrentUserUID = () => {
+  try {
+    return firebase.auth().currentUser.uid
+  } catch (error) {
+    throw new AuthenticationError('Not logged in.')
+  }
+}
+
+export default {
+  getCurrentUserUID
+}
```

Comment:
I changed this regarding this pull's files only. There may be some services (database and navigation) that still have some export default with object, changing them here would be messy as it affects SignUp/Login files. Can I do a separate pull for that only? (the same applies for require)

I left it as `import * as allExportsInAnObject from './path/to/file'`.

Otherwise it turned very hard to grasp interactions. For example:
`const imageURI = ImageService.getURIFromImage(image)`
`const imageReference = await StorageService.uploadImageAndReturnReference(imageURI)`
`await DatabaseService.addDiagnose(imageReference, problemDescription)`

would be:
`const imageURI = getURIFromImage(image)`
`const imageReference = await uploadImageAndReturnReference(imageURI)`
`await addDiagnose(imageReference, problemDescription)`

Even with descriptive names it becomes difficult to read. Then, I would have to replace `addDiagnose` to `addDiagnoseToDatabase` or maybe `submitDiagnose`, but reading it fast you are left wondering where it's submitted. Also without the `Service` part it gets difficult to see the separation of concerns. Let me know what you think!

Reply:
> I changed this regarding this pull's files only. There may be some services (database and navigation) that still have some export default with object, changing them here would be messy as it affects SignUp/Login files. Can I do a separate pull for that only? (the same applies for require)

You can leave it as-is for now, it can be changed in a future iteration.

> I left it as `import * as allExportsInAnObject from './path/to/file'`.

I agree, it's not a bad approach. What I didn't like was exporting an object by default because it removes the possibility to import each export individually.

> Even with descriptive names it becomes difficult to read. Then, I would have to replace addDiagnose to addDiagnoseToDatabase or maybe submitDiagnose, but reading it fast you are left wondering where it's submitted. Also without the Service part it gets difficult to see the separation of concerns.

The names can be changed with the `import` statement too. If I have a file where I only need one of your exports and I want a more descriptive name, I could write:
```javascript
import {
  uploadImageAndReturnReference as uploadImageToStorage
} from './path/to/file/about/storage'
```

Although the names of the functions should be as descriptive as possible, I don't know about the `AndReturnReference` in the example above. I'd almost always expect a CRUD operation to return a reference to the created/updated object. Again, you can leave it as-is for now. No need to change the code right now.